public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx

```diff
@@ -33,59 +32,36 @@ export interface CloudWatchLogsQueryFieldProps
     Themeable2 {
   absoluteRange: AbsoluteTimeRange;
   onLabelsRefresh?: () => void;
-  ExtraFieldElement?: ReactNode;
+  ExtraFieldElement?: React.ReactNode;
   exploreId: ExploreId;
   query: CloudWatchLogsQuery;
 }
 
-const addPaddingToButton = css`
-  padding: 1px 4px;
-`;
-interface State {
-  hint:
-    | {
-        message: string;
-        fix: {
-          label: string;
-          action: () => void;
-        };
-      }
-    | undefined;
-}
-
-class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogsQueryFieldProps, State> {
-  state: State = {
-    hint: undefined,
-  };
-
-  plugins: Array<Plugin<Editor>>;
+export const CloudWatchLogsQueryField = (props: CloudWatchLogsQueryFieldProps) => {
+  const { query, datasource, onChange, onRunQuery, ExtraFieldElement, data } = props;
 
-  constructor(props: CloudWatchLogsQueryFieldProps, context: React.Context<any>) {
-    super(props, context);
+  const showError = data && data.error && data.error.refId === query.refId;
+  const cleanText = datasource.languageProvider ? datasource.languageProvider.cleanText : undefined;
```

Comment:
```suggestion
  const { cleanText } = datasource.languageProvider;
```

nit: a minor improvement over what was originally there